docs/madr/decisions/053-sni.md

```diff
@@ -0,0 +1,62 @@
+# SNI for MeshService, MeshExternalService etc.
+
+* Status: accepted
+
+## Context and Problem Statement
+
+Currently, the SNI format is `<kuma.io/service>{mesh=<mesh>,tag1=value1}`. For example:
+* `backend{mesh=demo}`
+* `backend{mesh=demo,version=2}`
+
+We need to introduce SNI format for `MeshService` and `MeshExternalService` etc.
+Current format does not work, because we need to differentiate between `kuma.io/service` and MeshService.
+Current format also has other problems:
+* It violates [SNI spec](https://datatracker.ietf.org/doc/html/rfc3546#page-8). It should be a valid hostname. Using characters like `,{/` is not valid.
+  Envoy does not complain about it right now, but it can bite us in the future.
+* It can be too long. See https://github.com/kumahq/kuma/issues/8915
+
+## Considered Options
+
+* New format
+
+## Decision Outcome
+
+Chosen option: "New format", because it solves existing problem and lets us differentiate MeshService from old service.
+
+## Pros and Cons of the Options
+
+### New format
+
+It's important to note that we never parse SNI. We only use it for multizone communication on zone proxies to associate to which service pass the data.
+It means that we can hash the data as long as it's consistent hashing.
+We cannot put tags in the original form because they can violate hostname requirements. For example `/` in `kuma.io/service` is not ok.
+
+The new format could look like this then:
+`<format-version><hash>.<resource-name>.<port>.<mesh-name>.<resource-type>`
+* `<format-version>` - (required) `a`. 1 character. It's so we can evolve format in the future.
+* `<hash>` - (required) 16 characters of fnv64. It's `FNV64(resourceName,mesh,type,optional-data)`. Optional data can be for example tags.
```

Comment:
I'm uneasy about this in the end. Let's use sha256 especially now that we have versions in worst case we can change the version and change the hashing algorithm if we find this too slow...

Reply:
why? not enough chars for a hash = hash collisions?